scripts/pythondistdeps.py

```diff
@@ -224,82 +302,43 @@ def normalize_name(name):
-            # - Notes from an attempted rewrite from pkg_resources to
-            #   importlib.metadata in 2020 can be found in the message of
-            #   the commit that added this line.
-            from pkg_resources import Distribution, FileMetadata, PathMetadata, Requirement, parse_version
-            dist_name = basename(f)
-            if isdir(f):
-                path_item = dirname(f)
-                metadata = PathMetadata(path_item, f)
-            else:
-                path_item = f
-                metadata = FileMetadata(f)
-            dist = Distribution.from_location(path_item, dist_name, metadata)
-            # Check if py_version is defined in the metadata file/directory name
+            dist = Distribution(f)
             if not dist.py_version:
-                # Try to parse the Python version from the path the metadata
-                # resides at (e.g. /usr/lib/pythonX.Y/site-packages/...)
-                import re
-                res = re.search(r"/python(?P<pyver>\d+\.\d+)/", path_item)
-                if res:
-                    dist.py_version = res.group('pyver')
-                else:
-                    warn("Version for {!r} has not been found".format(dist), RuntimeWarning)
-                    continue
-
-            # pkg_resources use platform.python_version to evaluate if a
-            # dependency is relevant based on environment markers [1],
-            # e.g. requirement `argparse;python_version<"2.7"`
-            #
-            # Since we're running this script on one Python version while
-            # possibly evaluating packages for different versions, we mock the
-            # platform.python_version function. Discussed upstream [2].
-            #
-            # [1] https://www.python.org/dev/peps/pep-0508/#environment-markers
-            # [2] https://github.com/pypa/setuptools/pull/1275
-            import platform
-            platform.python_version = lambda: dist.py_version
-            platform.python_version_tuple = lambda: tuple(dist.py_version.split('.'))
-
-            # This is the PEP 503 normalized name.
-            # It does also convert dots to dashes, unlike dist.key.
-            # See https://bugzilla.redhat.com/show_bug.cgi?id=1791530
-            normalized_name = normalize_name(dist.project_name)
+                warn("Version for {!r} has not been found".format(dist), RuntimeWarning)
+                continue
 
             if args.majorver_provides or args.majorver_provides_versions or \
                     args.majorver_only or args.legacy_provides or args.legacy:
                 # Get the Python major version
                 pyver_major = dist.py_version.split('.')[0]
             if args.provides:
                 # If egg/dist metadata says package name is python, we provide python(abi)
-                if dist.key == 'python':
+                if dist.legacy_normalized_name == 'python':
```

Comment:
```suggestion
                if dist.normalized_name == 'python':
```

This does the same, but reads easier.